presto-tests/src/test/java/io/prestosql/execution/TestSqlStageExecution.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.execution;
+
+import com.google.common.collect.ImmutableMap;
+import io.prestosql.spi.QueryId;
+import io.prestosql.testing.DistributedQueryRunner;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+import static io.prestosql.SessionTestUtils.TEST_SESSION;
+import static io.prestosql.execution.QueryState.RUNNING;
+import static io.prestosql.execution.StageState.CANCELED;
+import static io.prestosql.execution.StageState.FLUSHING;
+import static io.prestosql.execution.TestQueryRunnerUtil.createQuery;
+import static io.prestosql.execution.TestQueryRunnerUtil.createQueryRunner;
+import static io.prestosql.execution.TestQueryRunnerUtil.waitForQueryState;
+import static org.testng.Assert.assertEquals;
+
+public class TestSqlStageExecution {
+
+    private DistributedQueryRunner queryRunner;
+
+    @Test(timeOut = 30_000)
+    public void testFlushingState()
+            throws Exception
+    {
+        queryRunner = createQueryRunner(ImmutableMap.of("node-scheduler.include-coordinator", "false"));
```

Comment:
Do we need distributed runner for this test?

Reply:
No, we don't need a distributed runner for this test. Used it because `TestQueryRunnerUtil` provided a clean interface for `waitForQueryState` (without fetching results) - is any existing one with `LocalQueryRunner` too?